omc3/sbs_matcher.py

```diff
@@ -0,0 +1,233 @@
+                "{name} = {name} {sign} {value};\n"
+                .format(name=varname, sign=sign_correct, value=abs(value))
+            )
+            match_data.write(
+                "{name} = {name} {sign} {value};\n"
+                .format(name=varname, sign=sign, value=abs(value))
+            )
+
+
+class InputData():
+
+    def __init__(self, lhc_mode, match_path, minimize, matchers_list):
+        self.lhc_mode = lhc_mode
+        self.match_path = match_path
+        self.matchers = matchers_list
+        self.minimize = minimize
+
+    def _get_matchers_list(self, input_data):
+        raw_matchers_list = input_data["matchers"]
+        for matcher_name, matcher_data in raw_matchers_list.iteritems():
+            matcher_type = matcher_data["type"]
+            matcher_beam = matcher_data["beam"]
+            matcher_variables = matcher_data["variables"]
+            MatcherClass = MATCHER_TYPES.get(matcher_type, None)
+            if MatcherClass is None:
+                raise ValueError('Unknown matcher type: ' + matcher_type +
+                                 ' must be in: ' + str(MATCHER_TYPES.keys()))
+            self.matchers.append(
+                MatcherClass(self.lhc_mode, matcher_beam,
+                             matcher_name, matcher_data, matcher_variables,
+                             self.match_path)
+            )
+
+    def _check_and_assign_attribute(self, input_data, attribute_name):
+        setattr(self, attribute_name, input_data[attribute_name])
+
+    # This transforms annoying unicode string into common byte string
+    @staticmethod
+    def _byteify(input_data):
+        if isinstance(input_data, dict):
+            return dict([(InputData._byteify(key), InputData._byteify(value))
+                         for key, value in input_data.iteritems()])
+        elif isinstance(input_data, list):
+            return [InputData._byteify(element) for element in input_data]
+        elif isinstance(input_data, str):
+            return input_data.encode('utf-8')
+        else:
+            return input_data
+
+
+def _run_gui(lhc_mode=None, match_path=None, input_dir=None):
+    try:
+        from sbs_general_matcher.gui import gui
+    except ImportError as err:
+        LOGGER.debug("ImportError importing GUI", exc_info=True)
+        LOGGER.info("Cannot start GUI using the current Python installation:")
+        LOGGER.info(str(err))
+        LOGGER.info("Launching OMC Anaconda Python...")
+        _run_gui_anaconda()
+        return
+    gui.main(lhc_mode, match_path, input_dir)
+
+
+def _run_gui_anaconda():
+    from subprocess import call
+    if sys.platform != "darwin":  # This is Mac
+        if "win" in sys.platform:
+            LOGGER.error("There is not Windows version of Anaconda in OMC.\
+                         Aborting.")
+            return
+    interpreter = os.path.join("/afs", "cern.ch", "eng", "sl", "lintrack",
+                               "omc3_python", "bin", "python")
```

Comment:
move to constant and put together in a file with acc-models repo/other external dependencies